solar/solar/core/resource/virtual_resource.py

```diff
@@ -108,10 +98,77 @@ def _get_template(name, content, kwargs, inputs):
             missing.append(input)
     if missing:
         raise Exception('[{0}] Validation error. Missing data in input: {1}'.format(name, missing))
-    template = Template(content)
+    template = Template(content, trim_blocks=True, lstrip_blocks=True)
     template = template.render(str=str, zip=zip, **kwargs)
     return template
 
 
 def is_virtual(path):
     return os.path.isfile(path)
+
+
+def parse_resources(resources):
+    created_resources = []
+    cwd = os.getcwd()
+    for r in resources:
+        connections = []
+        resource_name = r['id']
+        base_path = os.path.join(cwd, r['from'])
+        args = r['values']
+        new_resources = create(resource_name, base_path, args)
```

Comment:
For me `parse_` prefix in this function is misleading if you're creating something.

I'd split this into 2 functions: one for parsing `meta`, the other for creating resources. The one for parsing `meta` can be used later in the `load` function. BTW: can you add loading of VR too?